.azure-pipelines/testscripts_analyse/analyse_testcases.py

```diff
@@ -0,0 +1,175 @@
+Post-execution, the script will also append the scan time and track ID to the results.
+
+The return value is formatted as below:
+[
+    {
+        'testscript': 'acl/custom_acl_table/test_custom_acl_table.py',
+        'topology': 't0',
+        'trackid': '3aa57f0f-8f18-4cf7-ae1e-0a18973a0b86',
+        'scantime': '2024-05-31 06:53:40.826349',
+        'category': 'data',
+        'covered': False
+    },
+    {
+        'testscript': 'bgp/test_bgp_allow_list.py',
+        'topology': 't1',
+        'trackid': '3aa57f0f-8f18-4cf7-ae1e-0a18973a0b86',
+        'scantime': '2024-05-31 06:53:40.826349',
+        'category': 'control',
+        'covered': False
+    }
+]
+"""
+
+import yaml
+import re
+import os
+import sys
+import uuid
+import logging
+
+from natsort import natsorted
+from datetime import datetime
+from constant import DATAPLANE_FEATURES, PR_TOPOLOGY_TYPE, PR_TOPOLOGY_MAPPING
+
+
+def collect_all_scripts():
+    '''
+    This function collects all test scripts under the folder 'tests/'
+    and get the topology type marked in the script
+
+    The return value is a dict contains the script name and topology type
+    [{
+        'testscript': 'acl/custom_acl_table/test_custom_acl_table.py',
+        'topology': 't0'
+    }]
+    '''
+    location = sys.argv[1]
+
+    # Recursively find all files starting with "test_" and ending with ".py"
+    # Note: The full path and name of files are stored in a list named "files"
+    files = []
+    for root, dirs, file in os.walk(location):
+        for f in file:
+            if f.startswith("test_") and f.endswith(".py"):
+                files.append(os.path.join(root, f))
+    files = natsorted(files)
+
+    # Open each file and search for regex pattern
+    pattern = re.compile(r"[^@]pytest\.mark\.topology\(([^\)]*)\)")
+    test_scripts = []
+
+    for f in files:
+        # Remove prefix from file name:
+        filename = f[len(location) + 1:]
+        try:
+            with open(f, 'r') as file:
+                for line in file:
+                    # Get topology type of script from mark `pytest.mark.topology`
+                    match = pattern.search(line)
+                    if match:
+                        for topology in match.group(1).split(","):
+                            result = {
+                                "testscript": filename,
+                                "topology": topology.strip().strip('"').strip('\'')
+                            }
+                            test_scripts.append(result)
+        except Exception as e:
+            logging.error('Failed to load file {}, error {}'.format(f, e))
+
+    return test_scripts
+
+
+def get_testbed_type(topo_name):
+    pattern = re.compile(r'^(wan|t0|t1|ptf|fullmesh|dualtor|t2|tgen|mgmttor|m0|mc0|mx|dpu)')
+    match = pattern.match(topo_name)
+    if match is None:
+        logging.warning("Unsupported testbed type - {}".format(topo_name))
+        return "unsupported"
+    tb_type = match.group()
```

Comment:
Name of this variable needs improvement as well.

Reply:
Changed